### PR TITLE
sql: fix alter sequence statement reparse failures

### PR DESCRIPTION
**Pull request by e-mbrown**
Related to: #60722 

There were issues reparsing alter sequence when literals
were removed. Replacing the underscores with an int solved
the issue.

Release note: None

### CODE REVIEW

File: pkg/sql/sem/tree/hide_constants.go [hunk 1]
```diff
@@ -35,6 +35,9 @@ func (ctx *FmtCtx) formatNodeOrHideConstants(n NodeFormatter) {
 		case *Placeholder:
 			// Placeholders should be printed as placeholder markers.
 			// Deliberately empty so we format as normal.
+		case *NumVal:
```
Comment (rafiss) [inline pkg/sql/sem/tree/hide_constants.go:38]: hmm, looking through the test changes, it seems like this NumVal change didn't actually fix any of the cases. so let's actually take out the NumVal case and could you `-rewrite` the tests again? (this should also address some of the other test issues which seem to have come up with the change)

**Review by rafiss — CHANGES_REQUESTED** [review]

**Review by rafiss — CHANGES_REQUESTED** [review]
looking good! just need to get a few more cases

> Reviewed 1 of 27 files at r1, 2 of 26 files at r2.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/68251#-:-MfuX6Uf9gETGTChC98r:b-okgc12)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @e-mbrown)

___
*[pkg/sql/parser/testdata/alter_sequence, line 70 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/68251#-MfuWh6f69x-0Z0Uc_49:-MfuWh6f69x-0Z0Uc_4A:b12uwqe) ([raw file](https://github.com/cockroachdb/cockroach/blob/e53308e909e805bbf1880ab2221455718c686673/pkg/sql/parser/testdata/alter_sequence#L70)):*
> ```
> ALTER SEQUENCE IF EXISTS a NO CYCLE CACHE 1
> ALTER SEQUENCE IF EXISTS a NO CYCLE CACHE 1 -- fully parenthesized
> ALTER SEQUENCE IF EXISTS a NO CYCLE CACHE _ -- literals removed
> ```

looks like the CACHE option needs to be updated as well
___
*[pkg/sql/sem/tree/create.go, line 1520 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/68251#-MfuWubG39qzreGzA7oR:-MfuWubG39qzreGzA7oS:bi4mswe) ([raw file](https://github.com/cockroachdb/cockroach/blob/e53308e909e805bbf1880ab2221455718c686673/pkg/sql/sem/tree/create.go#L1520)):*
> ```Go
> 			// the cache option supports expressions.
> 			if ctx.flags.HasFlags(FmtHideConstants) {
> 				ctx.WriteByte('_')
> ```

this part needs to be updated too -- as well as any other SequenceOption where a number is turned into `_`
___
*[pkg/sql/sem/tree/create.go, line 1534 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/68251#-MfuX4Fw4I0pjnTnjQ6R:-MfuX4FxC2u5RWZR0jvW:b-a2xieb) ([raw file](https://github.com/cockroachdb/cockroach/blob/e53308e909e805bbf1880ab2221455718c686673/pkg/sql/sem/tree/create.go#L1534)):*
> ```Go
> 				// the min/max value options support expressions.
> 				if ctx.flags.HasFlags(FmtHideConstants) {
> 					ctx.WriteByte('_')
> ```

this one too

**Review by rafiss — APPROVED** [review]
this looks good to me!

when you're ready, you can trigger the merge by adding a `bors r+` (or `bors r=rafiss`) comment

**Comment by e-mbrown** [comment]
bors r=rafiss

**Pull request merged**